js/src/components/shipping-rate-section/minimum-order-card/minimum-order-card.js

```diff
@@ -0,0 +1,174 @@
+	);
+
+	const handleEditChange = ( oldGroup ) => ( newGroup ) => {
+		const newValue = value.map( ( shippingRate ) => {
+			const newShippingRate = {
+				...shippingRate,
+				options: {
+					...shippingRate.options,
+				},
+			};
+
+			if ( newGroup.countries.includes( newShippingRate.country ) ) {
+				/**
+				 * Shipping rate's country exists in the new value countries,
+				 * so we just assign the new value threshold.
+				 */
+				newShippingRate.options.free_shipping_threshold =
+					newGroup.threshold;
+			} else if (
+				oldGroup.countries.includes( newShippingRate.country )
+			) {
+				/**
+				 * Shipping rate's country does not exist in the new value countries,
+				 * but it exists in the old value countries.
+				 * This means users removed the country in the edit modal,
+				 * so we set the threshold value to undefined.
+				 */
+				newShippingRate.options.free_shipping_threshold = undefined;
+			}
+
+			return newShippingRate;
+		} );
+
+		onChange( newValue );
+	};
+
+	const handleAddChange = ( newGroup ) => {
+		const newValue = value.map( ( shippingRate ) => {
+			const newShippingRate = {
+				...shippingRate,
+				options: {
+					...shippingRate.options,
+				},
+			};
+
+			if ( newGroup.countries.includes( newShippingRate.country ) ) {
+				newShippingRate.options.free_shipping_threshold =
+					newGroup.threshold;
+			}
+
+			return newShippingRate;
+		} );
+
+		onChange( newValue );
+	};
+
+	const renderGroups = () => {
+		/**
+		 * If group length is 1, we render the group,
+		 * regardless of threshold is defined or not.
+		 */
+		if ( groups.length === 1 ) {
+			return (
+				<MinimumOrderInputControl
+					countryOptions={ countryOptions }
+					value={ groups[ 0 ] }
+					onChange={ handleEditChange( groups[ 0 ] ) }
+				/>
+			);
+		}
+
+		/**
+		 * When there is more than one group,
+		 * we render the groups with defined threshold first.
+		 * Then, if there is a group with undefined threshold,
+		 * we render an "Add another minimum order" button last.
+		 */
+		const emptyThresholdGroup = groups.find(
+			( group ) => group.threshold === undefined
+		);
+		const thresholdGroups = groups.filter(
+			( group ) => group !== emptyThresholdGroup
+		);
+
+		return (
+			<>
+				{ thresholdGroups.map( ( group ) => {
+					return (
+						<MinimumOrderInputControl
+							key={ group.countries.join( '-' ) }
+							countryOptions={ countryOptions }
+							value={ group }
+							onChange={ handleEditChange( group ) }
+						/>
+					);
+				} ) }
+				{ emptyThresholdGroup && (
+					<div>
+						<AppButtonModalTrigger
+							button={
+								<AppButton
+									isSecondary
+									icon={ <GridiconPlusSmall /> }
+								>
+									{ __(
+										'Add another minimum order',
+										'google-listings-and-ads'
+									) }
+								</AppButton>
+							}
+							modal={
+								<AddMinimumOrderModal
+									countryOptions={ countryOptions }
+									value={ emptyThresholdGroup }
```

Comment:
The inconsistency is between `<AddMinimumOrderModal>` and `<AddRateModal>`, not between `<EditRateModal>` and others.

Reply:
Gotcha! I have made it consistent with AddRateModal, in [`943ce4c` (#1266)](https://github.com/woocommerce/google-listings-and-ads/pull/1266/commits/943ce4c7500168e957e1651442262784fd9d4a42).

Side note: comparing Add modal and Edit modal, I do feel like it is inconsistent, I feel like we should make Add modal behave like Edit modal and make all the countries show up, because if all the countries can show up in Edit modal, then why don't they show up in Add modal? 🤔 For future iterations. 📜 😄

Reply:
> Side note: comparing Add modal and Edit modal, I do feel like it is inconsistent, I feel like we should make Add modal behave like Edit modal and make all the countries show up, because if all the countries can show up in Edit modal, then why don't they show up in Add modal?

Agree, I feel a bit concerned about the inconsistency. I didn't notice Add modal and Edit modal are using different country options until viewing this PR. 😂  And I couldn't recall if there's any specific purpose. So I tried to deduce the possibilities.

This is what I guessed when reading the codes:
- When editing, listing all countries make merchants easier to move countries from other groups into the current editing one, which reduces the many UI operations.
- When adding, listing remaining countries let merchants easier to see which countries are not yet set.

> For future iterations. 📜 😄

Regardless using all countries or remaining countries on which modal, I guess they both are reasonable. I also think we could go back later to see if it needs adjustment.